buffer/src/main/java/io/netty/buffer/PoolArena.java

```diff
@@ -724,5 +724,28 @@ protected void memoryCopy(ByteBuffer src, int srcOffset, ByteBuffer dst, int dst
                 dst.put(src);
             }
         }
+
+        @Override
+        protected void finalize() throws Throwable {
+            try {
+                super.finalize();
+            } finally {
+                destroyPoolSubPages(smallSubpagePools);
+                destroyPoolSubPages(tinySubpagePools);
+                destroyPoolChunkLists(qInit, q000, q025, q050, q075, q100);
```

Comment:
Perhaps you could define a protected method that calls the 3 `destroyPool*()` methods and leave the member fields `private`?